streams/src/main/java/org/apache/kafka/streams/processor/internals/GlobalStateManagerImpl.java

```diff
@@ -299,7 +318,17 @@ private void restoreState(final StateRestoreCallback stateRestoreCallback,
                             restoreRecords.add(recordConverter.convert(record));
                         }
                     }
-                    offset = globalConsumer.position(topicPartition);
+                    try {
```

Comment:
I can't comment above, but can the `poll` call itself throw a timeout exception? Or does it always just return no results in that case?

Reply:
> I can't comment above, but can the poll call itself throw a timeout exception?

No, it cannot. It would just return an empty result.

Reply:
Thanks!